### ISSUE TITLE
Invalid key error shown when viewing a post in the app

### ISSUE THREAD

**Issue by reginabally** [wordpress-mobile/WordPress-iOS#15611]
It was reported in 3617798-zen that the user wasn't able to view a post in the app through My Site > Blog Posts > View on a WordPress.com Simple site with a custom domain (Public). They saw the `Invalid key [1]` error on the screen with the URL `r-login.wordpress.com`. The same behavior was observed if the user previewed the post in the editor.

When they tried viewing the posts in the app on two other WordPress.com Simple sites (without custom domain) in private mode, they were asked to log in.

I wasn't able to replicate this in SSP.

### Expected behavior

The user expects to be able to view their posts in the app through My Site > Blog Posts > View.

### Actual behavior

The user wasn't able to view the post with either the error `Invalid key` error on their public site or they were asked to log in for the private sites.

### Steps to reproduce the behavior

1. Go to My Site > Blog Posts
2. Tap View on one of the posts
3. The screen prompted the error `Invalid key`
4. Tap the post to go to the editor
5. Tap three dots > Preview
6. The same issue in Step 3 is observed

![px93uvWz](https://user-images.githubusercontent.com/40906847/103973158-bf8ddb80-51a9-11eb-97b4-abe389fd6245.png)

##### Reported WPiOS 16.3.

I'm asking for the device and iOS version and will report back.

**Comment by startuptester** [comment]
I am not able to reproduce this on the latest beta. 
I will check back when further details are shared.

**Comment by reginabally** [comment]
The user writes back with the device information: iPhone SE (1st Gen), iOS 14.3.

They've also mentioned this:

> I tried an iPhone 7+ iOS 14.3, and in a split second, I see this link r-login.wordpress.com, the link I’m stuck with the SE, but goes straight to preview.

I'm asking them to try logging out from the app on iPhone SE and log back in to see if it helps.

**Comment by louinel** [comment]
I have another case 3672243-zen, I will ask the user more details about their device and update here.

**Comment by mdrockwell** [comment]
Another case here: 3758138-zen. I've asked them for some more device information.

**Comment by iamgabrielma** [comment]
> Another case here: 3758138-zen. I've asked them for some more device information.

The user replied with more info:

> iPhone 11, Version 14.4.
> iPad 6th generation, Version 14.4
> 
> I’ve noticed that posts etc don’t update on the other device if writing them between the two. I don’t do if often, but I have had to retrieve a couple of posts after losing it all / after publication.

**Comment by mdrockwell** [comment]
Another case here: 3766693-zd-woothemes

They have an iPhone 12 Pro Max

**Comment by reginabally** [comment]
Another report in 3879254-zd-woothemes. iPhone 7 Plus, iOS 14.4.2, WPiOS 16.9.

**Comment by mdrockwell** [comment]
Another here: 4134467-zd-woothemes

They're running 17.6, though. I asked them to update and let us know if they still have the issue afterward.

**Comment by reginabally** [comment]
Another report in 4137575-zd-woothemes.iPhone XR (iPhone11,8), iOS 14.1, WPiOS 17.6.0.5

**Comment by thabotswana** [comment]
> Another here: 4134467-zd-woothemes
> 
> They're running 17.6, though. I asked them to update and let us know if they still have the issue afterward.

The user said they would follow the instructions they were given later in the week. They did provide these details though:

Device: iPad Air 3 generation
iOS 14.6

**Comment by reginabally** [comment]
Another report in 4163453-zd-woothemes. iPhone 12 Pro Max, iOS 14.6, WPiOS 17.6.0.5.

**Comment by startuptester** [comment]
@chipsnyder I know it's not in the plan, but since this is rising quickly, can you please take a look as groundskeeper (or delegate it!) 🙏

**Comment by chipsnyder** [comment]
Yup I'll work on this one today 👍

**Comment by chipsnyder** [comment]
After a bit of investigation, I was able to recreate this issue.

After looking into the backend code I learned:
r-login will show this particular error if it can't determine the host from the re-login attempt or if the user isn't logged in. 

**Site:** Simple Site with a premium plan (I was able to recreate this when the site was private and public)
**Domain:** .blog domain aka hosted by Automattic. (I have a similar site with a domain hosted elsewhere and it seemed to work fine)

1. Add ~SMS~ Two-Step Authentication to your account 
2. Log in on the app or open the app (I was able to recreate this from either flow) 
    - 📓 SMS token requests were frequently failing silently for me https://github.com/wordpress-mobile/WordPress-iOS/issues/16436#issuecomment-885169481
3. Navigate to View Site
4. Notice the error.

I still need to investigate a fix for this but I was able to recreate the issue each time. 

**Edit** either 2FA option introduced this error for me
